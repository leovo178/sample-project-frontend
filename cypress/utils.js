/* eslint-disable no-undef */
const generateRandomNumber = (min, max) => {
	min = Math.ceil(min)
	max = Math.floor(max)
	return Math.floor(Math.random() * (max - min)) + min
}

const clickItemByTestId = testId =>
	cy
		.findByTestId(testId)
		.should('exist')
		.click()

const mockMessage = 'mockmessage'

export { generateRandomNumber, clickItemByTestId, mockMessage }
