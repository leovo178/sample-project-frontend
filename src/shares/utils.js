const AUTH_TOKEN = 'AUTH_TOKEN'

const getToken = () => localStorage.getItem(AUTH_TOKEN)
const setToken = token => localStorage.setItem(AUTH_TOKEN, token)
const deleteToken = () => localStorage.removeItem(AUTH_TOKEN)

const validateEmail = email => {
	const emailRegex = /^[a-z][a-z0-9_\.]{4,32}@[a-z0-9]{2,}(\.[a-z0-9]{2,4}){1,2}$/gm
	return email.match(emailRegex)
}

const validateName = name => {
	const nameRegex = /^[a-zA-Z0-9]{3,30}$/
	return name.match(nameRegex)
}

const validatePassword = password => {
	const passwordRegex = /^[a-zA-Z0-9]{3,}$/
	return password.match(passwordRegex)
}

export {
	getToken,
	setToken,
	deleteToken,
	validateEmail,
	validateName,
	validatePassword,
}
