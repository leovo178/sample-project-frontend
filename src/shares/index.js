import { getToken, setToken, deleteToken } from './utils'

export { getToken, setToken, deleteToken }
