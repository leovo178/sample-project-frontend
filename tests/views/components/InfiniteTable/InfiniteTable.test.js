import React from 'react'

import { cleanup, render } from '@testing-library/react'

import { InfiniteTable } from '@views_components'

import { mockUserList, mockUserTableHeader } from '@tests/shares/utils'

describe('InfiniteTable', () => {
	const mockProps = {
		items: mockUserList,
		columns: mockUserTableHeader,
		loadingMore: true,
		loadNextPage: jest.fn(),
	}

	const headerItems = mockProps.columns.map(item => item.headerVariable)

	afterEach(() => {
		cleanup()
	})

	it('should match snapshot', () => {
		const { container } = render(<InfiniteTable {...mockProps} />)

		expect(container).toMatchSnapshot()
	})

	it.each(headerItems)('should render %s header correctly', item => {
		const { getByTestId } = render(<InfiniteTable {...mockProps} />)
		const headerTestId = `tableheader-${item}`

		expect(getByTestId(headerTestId)).toBeTruthy()
	})
})
