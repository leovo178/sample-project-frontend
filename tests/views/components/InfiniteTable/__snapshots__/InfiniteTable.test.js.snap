// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteTable should match snapshot 1`] = `
<div
  style="width: 300px; height: 700px;"
>
  <div
    class="MuiBox-root MuiBox-root-5 makeStyles-root-1"
    data-testid="infinitetable"
  >
    <div
      class="MuiGrid-root makeStyles-container__header-2 MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
      >
        <p
          class="MuiTypography-root makeStyles-item__header-3 MuiTypography-body2"
          data-testid="tableheader-email"
        >
          EMAIL
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
      >
        <p
          class="MuiTypography-root makeStyles-item__header-3 MuiTypography-body2"
          data-testid="tableheader-name"
        >
          NAME
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-139 makeStyles-root-1"
    >
      <div
        style="overflow: visible; height: 0px; width: 0px;"
      >
        <div
          class="makeStyles-overflow-140"
          style="height: 700px; width: 200px; overflow: auto; position: relative; will-change: transform;"
        >
          <div
            style="height: 11900px; width: 100%;"
          >
            <div
              class="MuiGrid-root makeStyles-root-141 makeStyles-root-146 MuiGrid-container"
              data-testid="row-5e68995fb6d0bc05829b6e79"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
              >
                <div
                  class="MuiBox-root MuiBox-root-147 makeStyles-columns__item-144"
                >
                  <p
                    class="MuiTypography-root makeStyles-columns__item__typo-145 MuiTypography-body2"
                  >
                    steve@example.com
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
              >
                <div
                  class="MuiBox-root MuiBox-root-148 makeStyles-columns__item-144"
                >
                  <p
                    class="MuiTypography-root makeStyles-columns__item__typo-145 MuiTypography-body2"
                  >
                    90412
                  </p>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-root-141 makeStyles-root-149 MuiGrid-container"
              data-testid="row-5e68b5c3dd814dd832cda79d"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
              >
                <div
                  class="MuiBox-root MuiBox-root-150 makeStyles-columns__item-144"
                >
                  <p
                    class="MuiTypography-root makeStyles-columns__item__typo-145 MuiTypography-body2"
                  >
                    nvdai2355@gmail.com
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
              >
                <div
                  class="MuiBox-root MuiBox-root-151 makeStyles-columns__item-144"
                >
                  <p
                    class="MuiTypography-root makeStyles-columns__item__typo-145 MuiTypography-body2"
                  >
                    asdasdasd
                  </p>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-root-141 makeStyles-root-152 MuiGrid-container"
              data-testid="row-5e68b792dd814d8efacda7ad"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
              >
                <div
                  class="MuiBox-root MuiBox-root-153 makeStyles-columns__item-144"
                >
                  <p
                    class="MuiTypography-root makeStyles-columns__item__typo-145 MuiTypography-body2"
                  >
                    nvdai235556@gmail.com
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
              >
                <div
                  class="MuiBox-root MuiBox-root-154 makeStyles-columns__item-144"
                >
                  <p
                    class="MuiTypography-root makeStyles-columns__item__typo-145 MuiTypography-body2"
                  >
                    nvdai235556
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 201px; height: 701px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
  </div>
</div>
`;
